File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFlying.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPFlying implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "flying";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isFlying();
    }

    @Override
    public void restoreProperty(Player player, Boolean flying) {
        player.setFlying(flying);
    }

    @Override
    public Class<Boolean> getTypeClass() {
        return Boolean.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFlying.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPFlying extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "flying"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isFlying

  override def restoreProperty(player: Player, flying: Boolean): Unit =
    player.setFlying(flying)